unitTests/ui/SearchButtonTest.ts

```diff
@@ -27,5 +28,22 @@ export function SearchButtonTest() {
       test.cmp.click();
       expect(test.env.usageAnalytics.logSearchEvent).toHaveBeenCalledWith(analyticsActionCauseList.searchboxSubmit, {});
     });
+
+    it(`when the #options.searchbox is defined, when clicking the button,
+    it calls QueryStateModel #set to update the query`, () => {
+      const options: ISearchButtonOptions = {
+        searchbox: { getText: () => 'some query' }
+      };
+
+      test = Mock.basicComponentSetup<SearchButton>(SearchButton, options);
+      test.cmp.click();
+
+      expect(test.env.queryStateModel.set).toHaveBeenCalledWith(QueryStateModel.attributesEnum.q, options.searchbox.getText());
+    });
+
+    it(`when the #options is null, when clicking the button, it does not throw an error`, () => {
```

Comment:
Since this class is publicly exposed, clients could be passing `null` for the options to be able to specify the bindings. Note that prior to this PR, there were no options on this component. This unit test enforces the check for the truthy options on line 83 of `SearchButton.ts`.